tensornetwork/tests/network_test.py

```diff
@@ -537,3 +537,33 @@ def test_remove_node(backend):
   broken_edges_by_name, broken_edges_by_axis = tn.remove_node(b)
   assert broken_edges_by_name == {"0": a[0]}
   assert broken_edges_by_axis == {0: a[0]}
+
+
+@pytest.mark.parametrize("a, b, expected",
```

Comment:
please parametrize tests over different dtypes

Reply:
also check that dtypes are preserved (no implicit conversions)

Reply:
Add test cases. Please take another look.